src/primitives.jl

```diff
@@ -94,57 +110,48 @@ flipped_geom = GO.apply(GI.PointTrait, geom) do p
 end
 """
 @inline function apply(
-    f::F, ::Type{Target}, geom; calc_extent=false, threaded=false, kw...
-) where {F,Target}
+    f::F, target, geom; calc_extent=false, threaded=false, kw...
+) where F
     threaded = _booltype(threaded)
     calc_extent = _booltype(calc_extent)
-    _apply(f, Target, geom; threaded, calc_extent, kw...)
+    _apply(f, TraitTarget(target), geom; threaded, calc_extent, kw...)
 end
 
-#=
-We pass `threading` and `calc_extent` as types, not simple boolean values.  
-
-This is to help compilation - with a type to hold on to, it's easier for 
-the compiler to separate threaded and non-threaded code paths.
-=#
-struct _True end
-struct _False end
-
 @inline _booltype(x::Bool) = x ? _True() : _False()
 @inline _booltype(x::Union{_True,_False}) = x
```

Comment:
Can we move these definitions to where the booltypes are now defined, and change the last type from `Union` to `BoolsAsTypes`?